ydb/docs/en/core/integrations/goose.md

```diff
@@ -0,0 +1,128 @@
+```
+
+Further examples of calling `goose` commands will contain exactly this connection string.
+
+## Directory with migration files
+
+Let's create a migrations directory and then all `goose` commands should be executed in this directory:
+
+```
+$ mkdir migrations && cd migrations
+```
+
+## Adding a migration file
+
+The migration file can be generated using the `goose create` command:
+
+```
+$ goose ydb "grpc://localhost:2136/local?go_query_mode=scripting&go_fake_tx=scripting&go_query_bind=declare,numeric" create 00001_create_first_table sql
+2023/12/15 05:22:48 Created new file: 20231215052248_00001_create_table_users.sql
+```
+
+This means that the tool has created a new migration file `<timestamp>_00001_create_table_users.sql` where we can record the steps to change the schema for the YDB database, which is accessible through the corresponding connection string.
+
+So, after executing the goose create command, a migration file `<timestamp>_00001_create_table_users.sql` will be created with the following content:
+
+```
+-- +goose Up
+-- +goose StatementBegin
+SELECT 'up SQL query';
+-- +goose StatementEnd
+
+-- +goose Down
+-- +goose StatementBegin
+SELECT 'down SQL query';
+-- +goose StatementEnd
+```
+
+This migration file structure helps keep the instructions that lead to the next version of the database in context. It is also easy, without unnecessary distractions, to write instructions that roll back a database change.
+
+The migration file consists of two sections. The first is `+goose Up`, an area where we can record the migration steps. The second is `+goose Down`, an area in which we can write a step for invert making changes for the `+goose Up` steps. `Goose` carefully inserted placeholder queries:
+
+```
+SELECT 'up SQL query';
+```
+
+And
+
+```
+SELECT 'down SQL query';
+```
+
+so that we can instead enter essentially the migration requests themselves.
+
+Let's edit the migration file `<timestamp>_00001_create_table_users.sql` so that when applying the migration we create a table of the structure we need, and when rolling back the migration we delete the created table:
+
+```
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE users (
+     id Uint64,
+     username Text,
+     created_at TzDatetime,
+     PRIMARY KEY (id)
+);
+-- +goose StatementEnd
+
+-- +goose Down
+-- +goose StatementBegin
+DROP TABLE users;
+-- +goose StatementEnd
+```
+
+All subsequent migration files should be created in the same way.
+
+## Managing migrations
+
+The `goose` utility allows you to manage migrations via the command line:
+- `goose up` - apply all known migrations. For example, `goose ydb "grpc://localhost:2136/local?go_query_mode=scripting&go_fake_tx=scripting&go_query_bind=declare,numeric" up`.
+- `goose reset` - rollback all migrations. For example, `goose ydb "grpc://localhost:2136/local?go_query_mode=scripting&go_fake_tx=scripting&go_query_bind=declare,numeric" reset`.
```

Comment:
This sounds like an extremely dangerous command that could accidentally wipe the whole database (for example if you confuse it with `goose down`). Can we recommend some protection against that? Or at least explicitly warn readers.

Reply:
This is official command of goose. It is dangerous for all RDBMs, not only YDB
I can drop this command from article, but command `reset` - is a part of goose tool, and customer may faound it in `goose help`

Reply:
Ok, I will add warning about this command

Reply:
I added warning